test-utils/loadtester/src/main.rs

```diff
@@ -29,41 +39,160 @@ fn configure_logging(log_level: log::LevelFilter) {
+        2 => configure_logging(log::LevelFilter::Info),
+        _ => configure_logging(log::LevelFilter::Debug),
+    }
+
+    match matches.subcommand() {
+        ("create_genesis", Some(args)) => create_genesis(args),
+        ("run", Some(args)) => run(args),
+        _ => unreachable!(),
+    }
+}
+
+pub const CONFIG_FILENAME: &str = "config.json";
+
+fn create_genesis(matches: &clap::ArgMatches) {
+    let n = value_t_or_exit!(matches, "accounts", u64) as usize;
+    let v = value_t_or_exit!(matches, "validators", u64) as usize;
+    let prefix = value_t_or_exit!(matches, "prefix", String);
+    let dir_buf = value_t_or_exit!(matches, "home", PathBuf);
+    let dir = dir_buf.as_path();
+
+    let (mut configs, signers, network_signers, genesis_config) =
+        create_testnet_configs(v, n - v, &format!("{}.", prefix), false);
+    for i in 0..v {
+        let node_dir = dir.join(format!("{}.{}", prefix, i));
+        fs::create_dir_all(node_dir.clone()).expect("Failed to create directory");
+
+        signers[i].write_to_file(&node_dir.join(configs[i].validator_key_file.clone()));
+        network_signers[i].write_to_file(&node_dir.join(configs[i].node_key_file.clone()));
+
+        genesis_config.write_to_file(&node_dir.join(configs[i].genesis_file.clone()));
+        configs[i].consensus.min_num_peers = 0;
+        configs[i].write_to_file(&node_dir.join(CONFIG_FILENAME));
+        info!(target: "loadtester", "Generated node key, validator key, genesis file in {}", node_dir.to_str().unwrap());
+    }
+}
+
+fn run(matches: &clap::ArgMatches) {
+    let n = value_t_or_exit!(matches, "accounts", u64);
+    let prefix = value_t_or_exit!(matches, "prefix", String);
+    // let addr = value_t_or_exit!(matches, "addr", String);
+    let addrs = values_t_or_exit!(matches, "addrs", String);
+    let addr = &addrs[0];
+    let tps = value_t_or_exit!(matches, "tps", u64);
+    let duration = value_t_or_exit!(matches, "duration", u64);
+    let transaction_type = value_t_or_exit!(matches, "type", TransactionType);
+
+    let node = RemoteNode::new(SocketAddr::from_str(&addr).unwrap(), &[]);
+
+    // let peer_addrs = node.read().unwrap().peer_node_addrs().unwrap();
+    let peer_addrs = &addrs[1..];
+
+    let accounts: Vec<_> = (0..n).map(|i| format!("{}.{}", &prefix, i)).collect();
+
+    let num_nodes = peer_addrs.len() + 1;
     let accounts_per_node = accounts.len() / num_nodes;
-    let mut nodes = vec![];
-    for (i, addr) in addrs.iter().enumerate() {
+    node.write().unwrap().update_accounts(&accounts[0..accounts_per_node]);
+    let mut nodes = vec![node];
+    for (i, addr) in peer_addrs.iter().enumerate() {
         let node = RemoteNode::new(
             SocketAddr::from_str(addr).unwrap(),
-            &accounts[(i * accounts_per_node)..((i + 1) * accounts_per_node)],
+            &accounts[((i + 1) * accounts_per_node)..((i + 2) * accounts_per_node)],
         );
         nodes.push(node);
     }
```

Comment:
we should do canary testing by sending a tx to every single node and see that they got accepted.
using `add_transaction` that waits for tx acceptance.

Reply:
Temporarily have both add_txn (sync POST, async txn processing), add_txn_sync (sync POST, wait txn completed), and add_txn_async (async POST) for different use case